src/Http/Controllers/Traits/HasMedia.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Weap\Junction\Http\Controllers\Traits;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\Auth;
+use Spatie\MediaLibrary\MediaCollections\Models\Media;
+use Weap\Junction\Http\Utilities\MediaFile;
+use Weap\Junction\Models\MediaTemporaryUpload;
+
+trait HasMedia
+{
+    /**
+     * @param Model $model
+     * @param array<string, mixed> $validAttributes
+     * @return void
+     */
+    public function attachMedia(Model $model, array $validAttributes): void
+    {
+        if (! class_exists(Media::class) || ! config('media-library.media_model')) {
+            return;
+        }
+
+        foreach ($validAttributes as $key => $value) {
+            if (! is_array($value)) {
+                continue;
+            }
+
+            if ($key !== '_media') {
+                if ($model->$key instanceof Model) {
+                    $this->attachMedia($model->$key, $value);
+                }
+
+                continue;
+            }
+
+            if (! $this->isValidMediaArray($value)) {
+                continue;
+            }
+
+            foreach ($value as $collectionName => $uploadedFiles) {
+                /** @var MediaFile $uploadedFile */
+                foreach ($uploadedFiles as $uploadedFile) {
+                    $media = config('media-library.media_model')::findOrFail($uploadedFile->mediaId);
+
+                    abort_if($media->model_type !== MediaTemporaryUpload::class || Auth::user()->id !== $media->model->created_by_user_id, 404);
```

Comment:
- Might be bit cleaner to use Auth::id()
- Should this be 403 because it seems more like an authorization issue?